### PR TITLE
Move pkg/io to prow/io

### PR DESCRIPTION
**Pull request by alvaroaleman**
The package is only used by Prow, where I already am an approver. Alternatively, we could also move the package.

/assign @fejta @stevekuznetsov

### CODE REVIEW

File: pkg/io/OWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
```
Comment (fejta) [inline pkg/io/OWNERS:3]: Please add the other relevant approvers. This will make you the only person who ever gets assigned io changes.

Comment (alvaroaleman) [reply pkg/io/OWNERS:3]: Its definitely only used by prow:
```
$ ack -Q pkg/io -l|egrep -v '^prow'
BUILD.bazel
pkg/io/providers/providers_test.go
pkg/io/providers/BUILD.bazel
pkg/io/opener.go
pkg/io/BUILD.bazel
```

Do you mind if I instead just move the package under `prow/`?

Comment (fejta) [reply pkg/io/OWNERS:3]: SGTM

**Pull request merged**